engine/signer.go

```diff
@@ -0,0 +1,146 @@
+package engine
+
+import (
+	"context"
+	"fmt"
+
+	appchainClient "github.com/cosmos/cosmos-sdk/client"
+	cosmostx "github.com/cosmos/cosmos-sdk/client/tx"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+)
+
+type signer struct {
+	appchainCtx   *appchainClient.Context
+	privKey       *ed25519.PrivKey
+	pubKey        *cryptotypes.PubKey
+	address       *cryptotypes.Address
+	bech32Address *sdktypes.AccAddress
+}
+
+func NewSigner(appchainCtx *appchainClient.Context, privKey *ed25519.PrivKey) *signer {
+	return &signer{
+		appchainCtx: appchainCtx,
+		privKey:     privKey,
+	}
```

Comment:
curious - did you think about just setting the pubkey, address, and acc address during the `NewSigner` call instead of having them set as separate methods? Is there ever a case where we need some of those fields but not all of them?

Also, does `Address` need to be stored as a separate field or do we only care about `AccAddress` being stored on the `signer` struct? It looks like we can just pull the address when [signing](https://github.com/polymerdao/monomer/pull/123/files#diff-7873bc6b8c386aec42ec5b42d310a7e2afacfc9f1a1ca12e97abe4df06b7eafcR83) with `acc.GetAddress().String()`. Similarly, could the `PubKey` just be pulled off of the `privKey` with `privKey.PubKey()` whenever needed?

Reply:
Good points. `Address` is vestigial from the time I thought it was the actual field we needed. This stuff was cleaned up a little while `Signer` was moved into its own package.